sklearn/metrics/_classification.py

```diff
@@ -2089,55 +2130,128 @@ class after being classified as negative. This is the case when the
             f"problems, got targets of type: {y_type}"
         )
 
+    if raise_warning != "deprecated":
+        warnings.warn(
+            "`raise_warning` was deprecated in version 1.7 and will be removed "
+            "in 1.9, when an `UndefinedMetricWarning` will always raise in case of a "
+            "division by zero.",
+            FutureWarning,
+        )
+    else:
+        raise_warning = True
+        # TODO(1.9): This warning can be removed in 1.9. This is to make sure if users
+        # use the default args for `raise_warning` ("deprecated") and for
+        # `replace_undefined_by` (np.nan), they still get a FutureWarning telling them
+        # that the default return value will change:
+        if replace_undefined_by == "default":
+            warnings.warn(
+                "The default return value of `class_likelihood_ratios` in case of a "
+                "division by zero has been deprecated in 1.7 and will be changed to "
+                "the value set with the `replace_undefined_by` param in version 1.9. "
+                "Set `replace_undefined_by=np.nan` to silence this Warning.",
+                FutureWarning,
```

Comment:
It will be changed to the minimum score in version 1.9, at least this is how we had dealt with zero division up until now and this is how I have prepared this PR as well.

Not sure if this is an open question (https://github.com/scikit-learn/scikit-learn/issues/29048#issuecomment-2464875221) or if only I imagined it would be.